In [2]:
# film_genre_content.csv is a collection of different genre films queried from wikidata. 
# the code to abtain the content can be seen in "wiki_description_collector.ipynb"
import pandas as pd
df = pd.read_csv("wiki_genre_content.csv")
df.info(1)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 682 entries, 0 to 681
Data columns (total 15 columns):
 #   Column                           Non-Null Count  Dtype 
---  ------                           --------------  ----- 
 0   romantic comedy Q860626          657 non-null    object
 1   fantasy film Q157394             682 non-null    object
 2   dystopian films Q20443008        158 non-null    object
 3   children’s film Q2143665         105 non-null    object
 4   coming-of-age film (Q102429885)  174 non-null    object
 5   martial arts film (Q1033891)     423 non-null    object
 6   historical film                  161 non-null    object
 7   Adventure films                  199 non-null    object
 8   romantic comedy                  657 non-null    object
 9   fantasy                          682 non-null    object
 10  dystopian                        158 non-null    object
 11  historical                       161 non-null    object
 12  coming-of-age                    174

In [3]:
#create a new daraframe to set descriptions in seperate columns.
t = df[['romantic comedy', 'fantasy','historical','coming-of-age','wuxia','dystopian','Adventure']]

In [4]:
# set the sample sizes as same as 174.
t = t.dropna(thresh=5)
t.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 174 entries, 0 to 173
Data columns (total 7 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   romantic comedy  174 non-null    object
 1   fantasy          174 non-null    object
 2   historical       161 non-null    object
 3   coming-of-age    174 non-null    object
 4   wuxia            174 non-null    object
 5   dystopian        158 non-null    object
 6   Adventure        174 non-null    object
dtypes: object(7)
memory usage: 10.9+ KB


In [5]:
# for the following tokenization, use astype(str) to transform the value
t = pd.DataFrame(t[['romantic comedy', 'fantasy','historical','coming-of-age','wuxia','dystopian','Adventure']].astype(str))

In [6]:
# use package jieba for tokenization
import jieba
import re 


def cutword(x):
    seg = jieba.lcut(x, cut_all=False)
    return ' '.join(seg)


t['romantic_comedy'] = t['romantic comedy'].apply(cutword).astype(str)
t['fantasy'] = t['fantasy'].apply(cutword).astype(str)
t['historical'] = t['historical'].apply(cutword).astype(str)
t['coming-of-age'] = t['coming-of-age'].apply(cutword).astype(str)
t['wuxia'] = t['wuxia'].apply(cutword).astype(str)
t['dystopian'] = t['dystopian'].apply(cutword).astype(str)
t['Adventure'] = t['Adventure'].apply(cutword).astype(str)

Building prefix dict from the default dictionary ...
Loading model from cache /var/folders/qf/1_cfr6z54t5dv6mz1x_2zdcr0000gn/T/jieba.cache
Loading model cost 0.455 seconds.
Prefix dict has been built successfully.


In [7]:
t.head(2)

romantic comedy  \
0  《一半海水 一半火燄》（英語：ocean flame）是一部    年的電影，由星皓娱乐有限...   
1  《一吻定江山》（never been kissed）是一部    年由二十世紀福斯出品的美国...   

                                             fantasy  \
0  《 一万年 以后 》 （ 英語 ：             years   later ） ...   
1  《 歡樂谷 》 （ 英語 ： pleasantville ， 香港 译 《 一人 有 一點 ...   

                                          historical  \
0  《 一秒 鐘 》 是 一部         年 中國 大陆 劇情片 ， 由 张艺谋 執導 ，...   
1  《 法外 之王 》 （ 英語 ： outlaw   king ， 香港 译 《 逃犯 國王 ...   

                                       coming-of-age  \
0  《 一吻 定 江山 》 （ never   been   kissed ） 是 一部    ...   
1  一夫 當關 （ main   hoon   na ， 印地 語 ： म ै ं   ह ू ...   

                                               wuxia  \
0  《 一个 人 的 武林 》 （ 英語 ： kung   fu   jungle ） ， 香港...   
1  《 一代 宗師 》 （ 英語 ： the   grandmaster ） ， 於      ...   

                                           dystopian  \
0  《 一九八 四 》 （ 英語 ：         ） 是 一部         年 英國 黑...   
1  《 一九八 四 》 （ 英語 ： nineteen   eighty - four ， 又 ...   

                                           Adventure  \
0  《 x 戰警 ： 黑鳳凰 》 （ 英語 ： dark   phoenix ， 或 記作 x ...   
1  《 t -     ： 玩命 坦克 》   （ 俄語 ： т -     ） 是 由 ale...   

                                     romantic_comedy  
0  《 一半 海水   一半 火 燄 》 （ 英語 ： ocean   flame ） 是 一部...  
1  《 一吻 定 江山 》 （ never   been   kissed ） 是 一部    ...

In [8]:
# remove punctuation and stopwords 
# split tokenizaed text so that it can be calculated by length
t['romantic_comedy_splited'] = t['romantic_comedy'].str.split()
t['fantasy_splited'] = t['fantasy'].str.split()
t['historical_splited'] = t['historical'].str.split()
t['comingofage_splited'] = t['coming-of-age'].str.split()
t['wuxia_splited'] = t['wuxia'].str.split()
t['dystopian_splited'] = t['dystopian'].str.split()
t['adventure_splited'] = t['Adventure'].str.split()

#remove the punctuation chinese by using package zhon.
from zhon.hanzi import punctuation
punctuation_str = punctuation
for i in punctuation:
    t['romantic_comedy_splited'] = t['romantic_comedy_splited'].astype(str).str.replace(i, '')
    
for i in punctuation:
    t['fantasy_splited'] = t['fantasy_splited'].astype(str).str.replace(i, '')
for i in punctuation:
    t['historical_splited'] = t['historical_splited'].astype(str).str.replace(i, '')

for i in punctuation:
    t['comingofage_splited'] = t['comingofage_splited'].astype(str).str.replace(i, '')  
    
    
for i in punctuation:
    t['wuxia_splited'] = t['wuxia_splited'].astype(str).str.replace(i, '')  
for i in punctuation:
    t['dystopian_splited'] = t['dystopian_splited'].astype(str).str.replace(i, '')
for i in punctuation:
    t['adventure_splited'] = t['adventure_splited'].astype(str).str.replace(i, '')  

In [9]:
t.head(1)

romantic comedy  \
0  《一半海水 一半火燄》（英語：ocean flame）是一部    年的電影，由星皓娱乐有限...   

                                             fantasy  \
0  《 一万年 以后 》 （ 英語 ：             years   later ） ...   

                                          historical  \
0  《 一秒 鐘 》 是 一部         年 中國 大陆 劇情片 ， 由 张艺谋 執導 ，...   

                                       coming-of-age  \
0  《 一吻 定 江山 》 （ never   been   kissed ） 是 一部    ...   

                                               wuxia  \
0  《 一个 人 的 武林 》 （ 英語 ： kung   fu   jungle ） ， 香港...   

                                           dystopian  \
0  《 一九八 四 》 （ 英語 ：         ） 是 一部         年 英國 黑...   

                                           Adventure  \
0  《 x 戰警 ： 黑鳳凰 》 （ 英語 ： dark   phoenix ， 或 記作 x ...   

                                     romantic_comedy  \
0  《 一半 海水   一半 火 燄 》 （ 英語 ： ocean   flame ） 是 一部...   

                             romantic_comedy_splited  \
0  ['', '一半', '海水', '一半', '火', '燄', '', '', '英語',...   

                                     fantasy_splited  \
0  ['', '一万年', '以后', '', '', '英語', '', 'years', '...   

                                  historical_splited  \
0  ['', '一秒', '鐘', '', '是', '一部', '年', '中國', '大陆'...   

                                 comingofage_splited  \
0  ['', '一吻', '定', '江山', '', '', 'never', 'been',...   

                                       wuxia_splited  \
0  ['', '一个', '人', '的', '武林', '', '', '英語', '', '...   

                                   dystopian_splited  \
0  ['', '一九八', '四', '', '', '英語', '', '', '是', '一...   

                                   adventure_splited  
0  ['', 'x', '戰警', '', '黑鳳凰', '', '', '英語', '', '...

# love ratio compare : romantic comedy versus historical films

##First, calculate how many words a description has. Then calculate how many topic words a description has. Last, calculate the ratio of topic by ( topic words numbers / total length of description). T-test comparation conducted to compare the same dictionaries performance on different genre contents.

#### test wantsword method

In [10]:
#calculate the porpotion of love words in romantic comedy description and historical films' description

# the total length of each description 
t['romantic_comedy_splited_length'] = t['romantic_comedy_splited'].str.len()
t['historical_splited_length'] = t['historical_splited'].str.len()

# here we calculate the words numbers related to romantic love in wantswords
t['love_comedy'] = sum(t['romantic_comedy_splited'].str.count(x) for x in ("爱情","恋情","怜情","言情","恩爱","依恋","配","深情","钟情" ,"爱侣" ,"伴侣" ,"偕老","定终身","情爱","恋","爱上","痴情","动情","相思" ,"相爱","男欢女爱","儿女之情","一见钟情","情侣","心上人","才子佳人"))
t['love_historical'] = sum(t['historical_splited'].str.count(x) for x in ("爱情","恋情","怜情","言情","恩爱","依恋","配","深情","钟情" ,"爱侣" ,"伴侣" ,"偕老","定终身","情爱","恋","爱上","痴情","动情","相思" ,"相爱","男欢女爱","儿女之情","一见钟情","情侣","心上人","才子佳人"))

#caculate love elements in length, examples: t['love_xxxx'] / t['xxx_length']
t['love_ratio_comedy'] = t['love_comedy'] / t['romantic_comedy_splited_length']
t['love_ratio_historical'] = t['love_historical'] / t['historical_splited_length']

In [11]:
#check the mean numbers of topic in each genre (how many topic words one description has)
Total_1 = t['love_comedy'].mean()
Total_2 = t['love_historical'].mean()
print(Total_1)
print(Total_2)

0.9137931034482759
0.4827586206896552


In [12]:
# t - test for the topic ratio
from scipy.stats import ttest_ind
import seaborn as sns
ttest_ind(t['love_ratio_comedy'], t['love_ratio_historical'])

Ttest_indResult(statistic=4.050868889345594, pvalue=6.30342389566296e-05)

#### methord 2 bag of words

In [13]:
# here we calculate the words numbers related to romantic love based on method 2 
t['love_comedy_2'] = sum(t['romantic_comedy_splited'].str.count(x) for x in ('爱情', '情感', '甜蜜',  '真爱', '宿命', '挚爱','爱恋','虐恋', '深情',  '暗恋', '刻骨铭心', '真情', '恋爱', '爱情', '恋人', '情侣',  '初恋', '情人',  '暗恋', '初恋',  '烦恼',  '失恋', '欢喜','告白', '依恋'))
t['love_historical_2'] = sum(t['historical_splited'].str.count(x) for x in ('爱情', '情感', '甜蜜',  '真爱', '宿命', '挚爱','爱恋','虐恋', '深情',  '暗恋', '刻骨铭心', '真情', '恋爱', '爱情', '恋人', '情侣',  '初恋', '情人',  '暗恋', '初恋',  '烦恼',  '失恋', '欢喜','告白', '依恋'))

#calculate topic ratio
t['love_ratio_comedy_2'] = t['love_comedy_2'] / t['romantic_comedy_splited_length']
t['love_ratio_historical_2'] = t['love_historical_2'] / t['historical_splited_length']


ttest_ind(t['love_ratio_comedy_2'], t['love_ratio_historical_2'])

Ttest_indResult(statistic=3.584487145821982, pvalue=0.0003862486656242385)

In [16]:
Total_3 = t['love_comedy_2'].mean()
Total_4 = t['love_historical_2'].mean()
print(Total_3)
print(Total_4)

1.1494252873563218
0.1839080459770115


# fictionality compare : fantasy and romantic comedy 

In [17]:
# the total length of each description 
t['fantasy_splited_length'] = t['fantasy_splited'].str.len()

# here we calculate the words numbers related to topic
t['fictionality_historical'] = sum(t['historical_splited'].str.count(x) for x in ("虚构","虚造","杜撰","神话","神魔" ,"幻" ,"幻设" ,"想象" ,"臆造" ,"乌有","超现实" ,"神" ,"仙" ,"鬼" ,"怪" ,"妖", "魔","天堂" ,"地狱","穿越","科幻"))
t['fictionality_fantasy'] = sum(t['fantasy_splited'].str.count(x) for x in ("虚构","虚造","杜撰","神话","神魔" ,"幻" ,"幻设" ,"想象" ,"臆造" ,"乌有","超现实" ,"神" ,"仙" ,"鬼" ,"怪" ,"妖", "魔","天堂" ,"地狱","穿越","科幻"))


#caculate fictionality words in lenth: t['fictionality_xxxx'] / t['xxx_length']
t['fictionality_ratio_historical'] = t['fictionality_historical'] / t['historical_splited_length']
t['fictionality_ratio_fantasy'] = t['fictionality_fantasy'] / t['fantasy_splited_length']

ttest_ind(t['fictionality_ratio_fantasy'], t["fictionality_ratio_historical"])

Ttest_indResult(statistic=9.410298397669173, pvalue=7.102391249595409e-19)

In [18]:
Total_5 = t['fictionality_historical'].mean()
Total_6 = t['fictionality_fantasy'].mean()
print(Total_5)
print(Total_6)

1.1896551724137931
7.114942528735632


### methord 2 : bag of words for fictionality 

In [19]:
#repeat the steps
t['fictionality_historical_2'] = sum(t['historical_splited'].str.count(x) for x in (['虚构', '真实', '怪诞', '荒诞', '幻想','杜撰', '拟作', '编造',  '伪托', '假想', '魔剑',  '幽灵', '天使', '恶魔', '天堂', '魔鬼','战神', '虚幻', '荒诞', '奇异', '梦境', '超脱','幻想', '千变万化', '超现实','修仙',  '妖魔', '鬼魂', '妖怪', '封印', '魔界', '天界', '邪魔', '想象', '想像', '意象', '天马行空',  '虚幻', '梦幻', '另类', '梦境', '妖魔', '鬼魂', '妖怪', '封印', '魔界',  '化作',  '天界', '邪魔']))
t['fictionality_fantasy_2'] = sum(t['fantasy_splited'].str.count(x) for x in (['虚构', '真实', '怪诞', '荒诞', '幻想','杜撰', '拟作', '编造',  '伪托', '假想', '魔剑',  '幽灵', '天使', '恶魔', '天堂', '魔鬼','战神', '虚幻', '荒诞', '奇异', '梦境', '超脱','幻想', '千变万化', '超现实','修仙',  '妖魔', '鬼魂', '妖怪', '封印', '魔界', '天界', '邪魔', '想象', '想像', '意象', '天马行空',  '虚幻', '梦幻', '另类', '梦境', '妖魔', '鬼魂', '妖怪', '封印', '魔界',  '化作',  '天界', '邪魔']))


#caculate fictionality words in lenth: t['fictionality_xxxx'] / t['xxx_length']
t['fictionality_ratio_historical_2'] = t['fictionality_historical_2'] / t['historical_splited_length']
t['fictionality_ratio_fantasy_2'] = t['fictionality_fantasy_2'] / t['fantasy_splited_length']

ttest_ind(t['fictionality_ratio_fantasy_2'], t["fictionality_ratio_historical_2"])

Ttest_indResult(statistic=1.0900744051590328, pvalue=0.27643895045642075)

In [21]:
Total_7 = t['fictionality_historical_2'].mean()
Total_8 = t['fictionality_fantasy_2'].mean()
print(Total_7)
print(Total_8)

0.5
0.735632183908046


# morality compare: wuxia vs non-wuxia genre

### convert other genres into one column and randomly choose 173 wuxia films

In [25]:
#combine all non-wuxia genre into one column for compare 
non_wuxia = t[["romantic_comedy_splited","fantasy_splited","comingofage_splited"]].copy()
non_wuxia = pd.Series(non_wuxia.values.ravel('F')).to_frame()
#randomly choose 174 samples from wuxia films
non_wuxia = non_wuxia.sample(n=174) 
non_wuxia.columns = ['non_wuxia_splited']
non_wuxia.reset_index(drop=True)

non_wuxia_splited
0    ['', '善惡', '魔法', '學院', '', '', '英語', '', 'the'...
1    ['', '愛情', '的', '證明', '', '', '天城', '體', '', '...
2    ['', '布朗', '克斯', '的', '故事', '', '', '英語', '', ...
3    ['', '心靈', '訪客', '', '', '英語', '', 'finding', ...
4    ['', '小曼哈頓', '', '', '英語', '', 'little', 'manh...
..                                                 ...
169  ['', '大盜', '五右', '衛門', '', '或譯俠', '盜', '石川', '...
170  ['', '恋恋', '模范生', '', '', '英語', '', 'but', 'i'...
171  ['', '超', '能', '失控', '', '', '英語', '', 'chroni...
172  ['', '卡美拉', '雷吉翁', '襲來', '', '', '原題', '', 'ガ'...
173  ['', '最後一場', '電影', '', '', '英語', '', 'the', 'l...

[174 rows x 1 columns]

In [26]:
#create new dataframe for wuxia and non_wuxia
wuxia = t['wuxia_splited'].copy()
wuxia = wuxia.to_frame()
wuxia

wuxia_splited
0    ['', '一个', '人', '的', '武林', '', '', '英語', '', '...
1    ['', '一代', '宗師', '', '', '英語', '', 'the', 'gra...
2    ['', '一個', '好人', '', '', '英語', '', 'mr', '.', ...
3    ['', '點止', '功夫', '咁', '簡單', '', '', '英語', '', ...
4    ['', '一眉', '道人', '', '', '英語', '', 'vampire', ...
..                                                 ...
169  ['', '新獨臂', '刀', '', '', '英語', '', 'the', 'new...
170  ['', '新', '精武', '門', '', '', '英語', '', 'fist',...
171  ['', '新', '精武门', '', '', '英語', '', 'new', 'fis...
172  ['', '新', '紮', '师兄', '', '', '英語', '', 'moving...
173  ['', '新', '蜀山', '剑侠', '', '', '英語', '', 'zu', ...

[174 rows x 1 columns]

### wantsword dictionary morality

In [27]:
from scipy.stats import ttest_ind
import seaborn as sns

#repeat the steps 
wuxia['wuxia_splited_length'] =wuxia['wuxia_splited'].str.len()
non_wuxia['non_wuxia_splited_length'] = non_wuxia['non_wuxia_splited'].str.len()


wuxia['morality_wuxia'] = sum(wuxia['wuxia_splited'].str.count(x) for x in ("道德","德操",'德行','德性','仁义','公德','道义','伦理','仁','美德','明德','良知','义理','正义','公允','公平','正直','公正','公道','守正不阿','合理','公正廉明','正义','公明正大','秉公','铁面无私','大义','义举','道义','仁义','仗义','合理','愧疚','抱愧','疚愧','负疚','内疚','感愧','于心有愧','愤愤不平','愤懑','气愤','愤愤','郁愤','忿忿不平','悲愤','气不平','归咎','归咎于','值得'))
non_wuxia['morality_non_wuxia'] = sum(non_wuxia['non_wuxia_splited'].str.count(x) for x in ("道德","德操",'德行','德性','仁义','公德','道义','伦理','仁','美德','明德','良知','义理','正义','公允','公平','正直','公正','公道','守正不阿','合理','公正廉明','正义','公明正大','秉公','铁面无私','大义','义举','道义','仁义','仗义','合理','愧疚','抱愧','疚愧','负疚','内疚','感愧','于心有愧','愤愤不平','愤懑','气愤','愤愤','郁愤','忿忿不平','悲愤','气不平','归咎','归咎于','值得'))

wuxia['morality_ratio_wuxia'] = wuxia['morality_wuxia'] / wuxia['wuxia_splited_length']
non_wuxia['morality_ratio_non_wuxia'] = non_wuxia['morality_non_wuxia'] / non_wuxia['non_wuxia_splited_length']

#t-test
ttest_ind(wuxia['morality_ratio_wuxia'], non_wuxia["morality_ratio_non_wuxia"])

Ttest_indResult(statistic=2.768714862108022, pvalue=0.0059309352208403165)

In [28]:
Total_9 = wuxia['morality_wuxia'].mean()
Total_10 = non_wuxia['morality_non_wuxia'].mean()
print(Total_9)
print(Total_10)

0.6206896551724138
0.16091954022988506


### method 2 :bag of words for morality

In [29]:

wuxia['morality_wuxia_2'] = sum(wuxia['wuxia_splited'].str.count(x) for x in ('道德', '伦理道德', '道义','良知', '德性','天道', '人道', '良知',  '公正', '公平', '诚信','平等',  '正义','公平正义',  '天理', '礼义',  '大义', '忠义','内疚', '羞愧','自责','愧疚', '内疚','愤慨', '愤懑', '怨愤', '激愤',  '不平', '悲愤', '激愤',  '愤懑','义愤填膺','归咎'))
non_wuxia['morality_non_wuxia_2'] = sum(non_wuxia['non_wuxia_splited'].str.count(x) for x in ('道德', '伦理道德', '道义','良知', '德性','天道', '人道', '良知',  '公正', '公平', '诚信','平等',  '正义','公平正义',  '天理', '礼义',  '大义', '忠义','内疚', '羞愧','自责','愧疚', '内疚','愤慨', '愤懑', '怨愤', '激愤',  '不平', '悲愤', '激愤',  '愤懑','义愤填膺','归咎'))


#caculate fictionality words in lenth: t['fictionality_xxxx'] / t['xxx_length']
wuxia['morality_ratio_wuxia_2'] = wuxia['morality_wuxia_2'] / wuxia['wuxia_splited_length']
non_wuxia['morality_ratio_non_wuxia_2'] = non_wuxia['morality_non_wuxia_2'] / non_wuxia['non_wuxia_splited_length']

ttest_ind(wuxia['morality_ratio_wuxia_2'], non_wuxia["morality_ratio_non_wuxia_2"])

Ttest_indResult(statistic=1.5215874660792346, pvalue=0.12902582231000181)

In [30]:
Total_11 = wuxia['morality_wuxia_2'].mean()

Total_12 = non_wuxia['morality_non_wuxia_2'].mean()
print(Total_11)
print(Total_12)

0.12643678160919541
0.09195402298850575


# prosociality compare: wuxia versus non_wuxia genre


### wantsword method: prosociality (friendship)

In [31]:
from scipy.stats import ttest_ind
import seaborn as sns

wuxia['wuxia_splited_length'] =wuxia['wuxia_splited'].str.len()
non_wuxia['non_wuxia_splited_length'] = non_wuxia['non_wuxia_splited'].str.len()

wuxia['prosociality_wuxia'] = sum(wuxia['wuxia_splited'].str.count(x) for x in ('朋友','友谊','同袍','知己','知心','结拜','知音','肺腑之交','心心相印','志同道合','推心置腹','肝胆相照','忠肝义胆','结义','结契','兄弟','哥们儿','青梅','竹马','同情','怜恤','恻隐','心软','怜惜','怜','体贴','帮补','援手','舍己为人','与人为善','帮助','相助','救','施恩' ,'济人利物','匡扶','扶持','支持','施予' ,'关怀','关爱', '关照', '关心','扶贫济困','扶危济困','体恤','慈善','行好','热心肠','善心','仁','分享' ,'同享','安慰','抚慰','慰' ,'安抚'))
non_wuxia['prosociality_non_wuxia'] = sum(non_wuxia['non_wuxia_splited'].str.count(x) for x in ('朋友','友谊','同袍','知己','知心','结拜','知音','肺腑之交','心心相印','志同道合','推心置腹','肝胆相照','忠肝义胆','结义','结契','兄弟','哥们儿','青梅','竹马','同情','怜恤','恻隐','心软','怜惜','怜','体贴','帮补','援手','舍己为人','与人为善','帮助','相助','救','施恩' ,'济人利物','匡扶','扶持','支持','施予' ,'关怀','关爱', '关照', '关心','扶贫济困','扶危济困','体恤','慈善','行好','热心肠','善心','仁','分享' ,'同享','安慰','抚慰','慰' ,'安抚'))


wuxia['prosociality_ratio_wuxia'] = wuxia['prosociality_wuxia'] / wuxia['wuxia_splited_length']
non_wuxia['prosociality_ratio_non_wuxia'] = non_wuxia['prosociality_non_wuxia'] / non_wuxia['non_wuxia_splited_length']

ttest_ind(wuxia['prosociality_ratio_wuxia'], non_wuxia["prosociality_ratio_non_wuxia"])

Ttest_indResult(statistic=5.081863076808856, pvalue=6.129099777536459e-07)

In [32]:
Total_13 = wuxia['prosociality_wuxia'].mean()

Total_14 = non_wuxia['prosociality_non_wuxia'].mean()
print(Total_13)
print(Total_14)

2.0
1.3275862068965518


### method 2 : bag of words for prosociality (friendship)

In [33]:

wuxia['prosociality_wuxia_2'] = sum(wuxia['wuxia_splited'].str.count(x) for x in ('关心', '尊重', '关切', '热衷', '热心', '爱护', '同情', '照顾', '同情', '感激', '珍视', '诚挚', '担忧', '惋惜', '敬重', '感谢', '报答', '祝贺', '安慰', '悼念', '献给', '献上', '款待', '爱护', '体谅', '体恤', '敬爱', '帮助', '协助', '鼓励', '支持', '同伴', '激励', '协助', '筹划', '支援', '联络', '扶持', '召集', '老师', '同学', '师傅', '同事', '朋友', '导师', '朋友', '同学','好友', '友谊', '情谊','友情', '友爱', '伴侣', '同舟共济', '并肩', '互惠互利', '平等互利', '互惠', '相互尊重', '和而不同', '捐赠', '捐助', '捐款', '善款', '捐献', '捐给', '募集', '发放', '捐出', '筹集', '鼓励','怜悯'))
non_wuxia['prosociality_non_wuxia_2'] = sum(non_wuxia['non_wuxia_splited'].str.count(x) for x in ('关心', '尊重', '关切', '热衷', '热心', '爱护', '同情', '照顾', '同情', '感激', '珍视', '诚挚', '担忧', '惋惜', '敬重', '感谢', '报答', '祝贺', '安慰', '悼念', '献给', '献上', '款待', '爱护', '体谅', '体恤', '敬爱', '帮助', '协助', '鼓励', '支持', '同伴', '激励', '协助', '筹划', '支援', '联络', '扶持', '召集', '老师', '同学', '师傅', '同事', '朋友', '导师', '朋友', '同学','好友', '友谊', '情谊','友情', '友爱', '伴侣', '同舟共济', '并肩', '互惠互利', '平等互利', '互惠', '相互尊重', '和而不同', '捐赠', '捐助', '捐款', '善款', '捐献', '捐给', '募集', '发放', '捐出', '筹集', '鼓励','怜悯'))


wuxia['prosociality_ratio_wuxia_2'] = wuxia['prosociality_wuxia_2'] / wuxia['wuxia_splited_length']
non_wuxia['prosociality_ratio_non_wuxia_2'] = non_wuxia['prosociality_non_wuxia_2'] / non_wuxia['non_wuxia_splited_length']

ttest_ind(wuxia['prosociality_ratio_wuxia_2'], non_wuxia["prosociality_ratio_non_wuxia_2"])

Ttest_indResult(statistic=0.6997480998861823, pvalue=0.4845548132075811)

In [35]:
Total_15 = wuxia['prosociality_wuxia_2'].mean()
Total_16 = non_wuxia['prosociality_non_wuxia_2'].mean()
print(Total_15)
print(Total_16)

1.0804597701149425
1.235632183908046


# kinship compare: romantic comedy vs adventure films

### wantsword method for kinship

In [36]:
from scipy.stats import ttest_ind
import seaborn as sns
t['adventure_splited_length'] = t['adventure_splited'].str.len()
t['romantic_comedy_splited_length'] = t['romantic_comedy_splited'].str.len()

t['kinship_comedy'] = sum(t['romantic_comedy_splited'].str.count(x) for x in ('亲戚','血亲','直系亲属','亲戚','姻亲','血缘','直系','嫡亲','亲族','亲人','世亲','亲眷','子','嫡堂','血统','亲子','宗亲','族亲','胞族','干亲','嫡','亲缘','姻亚','外族','祖宗','父系','宗族','近亲','尊亲','母系','嫡系','六亲','同胞','家庭','亲人','父母','家长'))
t['kinship_adventure'] = sum(t['adventure_splited'].str.count(x) for x in ('亲戚','血亲','直系亲属','亲戚','姻亲','血缘','直系','嫡亲','亲族','亲人','世亲','亲眷','子','嫡堂','血统','亲子','宗亲','族亲','胞族','干亲','嫡','亲缘','姻亚','外族','祖宗','父系','宗族','近亲','尊亲','母系','嫡系','六亲','同胞','家庭','亲人','父母','家长'))


t['kinship_ratio_comedy'] = t['kinship_comedy'] / t['romantic_comedy_splited_length']
t['kinship_ratio_adventure'] = t['kinship_adventure'] / t['adventure_splited_length']

ttest_ind(t['kinship_ratio_comedy'], t['kinship_ratio_adventure'])

Ttest_indResult(statistic=0.8077953054006652, pvalue=0.41976342838115954)

In [37]:
Total_17 = t['kinship_ratio_comedy'].mean()
Total_18 = t['kinship_ratio_adventure'].mean()
print(Total_17)
print(Total_18)

0.0007091284143417267
0.0006152676116488727


### method 2: bag of words for kinship

In [38]:

t['kinship_comedy_2'] = sum(t['romantic_comedy_splited'].str.count(x) for x in ('亲戚', '舅舅', '亲戚朋友', '老乡', '同乡', '表弟', '亲朋好友', '熟人', '女婿', '妹夫', '远房亲戚', '舅父', '侄女', '姑姑', '岳母','家族','大家族', '后裔', '王室', '后代', '氏族', '皇室', '旁支', '表亲', '王族', '母系', '父祖', '表兄弟', '堂兄弟','父母', '双亲', '父母亲', '母亲', '家人', '丈夫', '子女', '养父母', '父亲', '祖母', '小孩', '继父', '儿女', '爸妈', '儿媳','孝', '忠', '恭', '贞', '昭', '懿', '贤', '世子','养育', '抚育', '抚养', '扶养', '养大', '收养', '哺育', '养', '领养', '供养', '照看', '教养', '服侍', '生养', '照管','血缘', '血亲', '父系', '同宗', '母系', '亲密关系', '母方', '同族', '亲兄弟', '亲子关系', '谱系', '该族','族', '部落', '部族', '族群'))
t['kinship_adventure_2'] = sum(t['adventure_splited'].str.count(x) for x in ('亲戚', '舅舅', '亲戚朋友', '老乡', '同乡', '表弟', '亲朋好友', '熟人', '女婿', '妹夫', '远房亲戚', '舅父', '侄女', '姑姑', '岳母','家族','大家族', '后裔', '王室', '后代', '氏族', '皇室', '旁支', '表亲', '王族', '母系', '父祖', '表兄弟', '堂兄弟','父母', '双亲', '父母亲', '母亲', '家人', '丈夫', '子女', '养父母', '父亲', '祖母', '小孩', '继父', '儿女', '爸妈', '儿媳','孝', '忠', '恭', '贞', '昭', '懿', '贤', '世子','养育', '抚育', '抚养', '扶养', '养大', '收养', '哺育', '养', '领养', '供养', '照看', '教养', '服侍', '生养', '照管','血缘', '血亲', '父系', '同宗', '母系', '亲密关系', '母方', '同族', '亲兄弟', '亲子关系', '谱系', '该族','族', '部落', '部族', '族群'))


t['kinship_ratio_comedy_2'] = t['kinship_comedy_2'] / t['romantic_comedy_splited_length']
t['kinship_ratio_adventure_2'] = t['kinship_adventure_2'] / t['adventure_splited_length']

ttest_ind(t['kinship_ratio_comedy_2'], t['kinship_ratio_adventure_2'])

Ttest_indResult(statistic=0.3095005171363655, pvalue=0.7571271230737047)

In [39]:
Total_19 = t['kinship_ratio_comedy_2'].mean()

Total_20 = t['kinship_ratio_adventure_2'].mean()
print(Total_19)
print(Total_20)

0.00038541342209396577
0.0003534966266440119


# status and conflict compare: historical versus fantasy

### method wantsword for status and conflict

In [40]:

# the total length of each description 
t['historical_splited_length'] = t['historical_splited'].str.len()
t['romantic_comedy_splited_length'] = t['romantic_comedy_splited'].str.len()


t['status_comedy'] = sum(t['romantic_comedy_splited'].str.count(x) for x in ('身份','名分','出身','位分','资格','体面','排场','身价','官身','籍','身段','位望','位居','层级','上流','身价','份儿','霸权','名位','声价','势力','待遇','阶层','贵贱','权位'))
t['status_historical'] = sum(t['historical_splited'].str.count(x) for x in ('身份','名分','出身','位分','资格','体面','排场','身价','官身','籍','身段','位望','位居','层级','上流','身价','份儿','霸权','名位','声价','势力','待遇','阶层','贵贱','权位'))


t['status_ratio_comedy'] = t['status_comedy'] / t['romantic_comedy_splited_length']
t['status_ratio_historical'] = t['status_historical'] / t['historical_splited_length']

ttest_ind(t['status_ratio_historical'],t['status_ratio_comedy'])

Ttest_indResult(statistic=0.5774247756691117, pvalue=0.5640280807254625)

In [41]:
Total_25 = t['status_comedy'].mean()
Total_26 = t['status_historical'].mean()
print(Total_25)
print(Total_26)

0.1839080459770115
0.4425287356321839


### method 2 bag of words for status and conflict

In [42]:

t['status_comedy_2'] = sum(t['romantic_comedy_splited'].str.count(x) for x in ('地位','声望', '竞争', '主导',  '威望', '声誉', '重要性','威信','品第', '王侯', '官', '身分', '身份', '名号','荣誉', '头衔', '功绩', '骄傲', '自豪', '钦佩', '敬佩','光荣','崇敬', '仰慕','服从', '顺从', '遵从', '命令', '遵守', '听从', '效忠', '听命', '顺服', '依从', '听命于', '遵照', '忠于','权威', '权威性', '实力', '竞争', '战力', '攻击', '统治',  '战斗','羞耻','厌恶', '轻蔑', '可耻',  '允许', '容许', '准许', '禁止', '限制', '强制', '强迫',  '严禁', '明令禁止', '禁令',  '不准', '禁用', '不容许', '准许', '强迫', '容许','义务', '权利', '责任', '职责', '冲突', '纷争', '暴力', '争执', '流血', '关系紧张', '矛盾', '争斗', '对立', '动乱', '暴乱','战争', '内战', '战事', '叛乱', '起义', '叛变', '起事', '反叛', '兵变', '暴乱', '暴动', '民变', '农民起义', '反抗', '动乱', '内乱', '反乱', '叛军'))
t['status_historical_2'] = sum(t['historical_splited'].str.count(x) for x in ('地位','声望', '竞争', '主导',  '威望', '声誉', '重要性','威信','品第', '王侯', '官', '身分', '身份', '名号','荣誉', '头衔', '功绩', '骄傲', '自豪', '钦佩', '敬佩','光荣','崇敬', '仰慕','服从', '顺从', '遵从', '命令', '遵守', '听从', '效忠', '听命', '顺服', '依从', '听命于', '遵照', '忠于','权威', '权威性', '实力', '竞争', '战力', '攻击', '统治',  '战斗','羞耻','厌恶', '轻蔑', '可耻',  '允许', '容许', '准许', '禁止', '限制', '强制', '强迫',  '严禁', '明令禁止', '禁令',  '不准', '禁用', '不容许', '准许', '强迫', '容许','义务', '权利', '责任', '职责', '冲突', '纷争', '暴力', '争执', '流血', '关系紧张', '矛盾', '争斗', '对立', '动乱', '暴乱','战争', '内战', '战事', '叛乱', '起义', '叛变', '起事', '反叛', '兵变', '暴乱', '暴动', '民变', '农民起义', '反抗', '动乱', '内乱', '反乱', '叛军'))


t['status_ratio_comedy_2'] = t['status_comedy_2'] / t['romantic_comedy_splited_length']
t['status_ratio_historical_2'] = t['status_historical_2'] / t['historical_splited_length']

ttest_ind(t['status_ratio_historical_2'],t['status_ratio_comedy_2'])

Ttest_indResult(statistic=3.71968003870418, pvalue=0.00023266142505618957)

In [43]:
Total_27 = t['status_comedy_2'].mean()

Total_28 = t['status_historical_2'].mean()
print(Total_27)
print(Total_28)

0.5574712643678161
2.281609195402299


# self-development compare: coming of age film versus romantic comedy film

### wantsword method for self development

In [45]:

# the total length of each description 
t['comingofage_splited_length'] = t['comingofage_splited'].str.len()


t['development_comingofage'] = sum(t['comingofage_splited'].str.count(x) for x in ('修身','慎行','修行','修养','修身养性','修身洁行','正身','立身行道','自修','修炼','自治','砥砺德行','修身齐家','修真养性','教养','自强','自发','自我表现','自足','自修','自持','自主','自制力','自矜'))
t['development_comedy'] = sum(t['romantic_comedy_splited'].str.count(x) for x in ('修身','慎行','修行','修养','修身养性','修身洁行','正身','立身行道','自修','修炼','自治','砥砺德行','修身齐家','修真养性','教养','自强','自发','自我表现','自足','自修','自持','自主','自制力','自矜'))


t['development_ratio_comingofage'] = t['development_comingofage'] / t['comingofage_splited_length']
t['development_ratio_comedy'] = t['development_comedy'] / t['romantic_comedy_splited_length']

ttest_ind(t['development_ratio_comingofage'], t['development_ratio_comedy'])

Ttest_indResult(statistic=-0.9929286662607051, pvalue=0.3214385939167695)

In [46]:
Total_29 = t['development_comingofage'].mean()

Total_30 = t['development_comedy'].mean()
print(Total_29)
print(Total_30)

0.005747126436781609
0.028735632183908046


### method 2 bag of words for self development

In [48]:

# the total length of each description 
t['comingofage_splited_length'] = t['comingofage_splited'].str.len()


t['development_comingofage_2'] = sum(t['comingofage_splited'].str.count(x) for x in ('成长',  '蜕变',  '茁壮成长', '茁壮', '健康成长','修炼',  '修行', '修道', '修持', '苦行', '苦修', '参悟','个性', '性格', '气质', '学习', '自学', '研习', '学', '努力', '讲授', '钻研', '读书',  '研读', '进修', '授课', '修习','自律', '自我管理','自律','修身养性', '陶冶','自强', '力行', '有志', '忠义',  '自力', '自强'))
t['development_comedy_2'] = sum(t['romantic_comedy_splited'].str.count(x) for x in ('成长',  '蜕变',  '茁壮成长', '茁壮', '健康成长','修炼',  '修行', '修道', '修持', '苦行', '苦修', '参悟','个性', '性格', '气质', '学习', '自学', '研习', '学', '努力', '讲授', '钻研', '读书',  '研读', '进修', '授课', '修习','自律', '自我管理','自律','修身养性', '陶冶','自强', '力行', '有志', '忠义',  '自力', '自强'))


t['development_ratio_comingofage_2'] = t['development_comingofage_2'] / t['comingofage_splited_length']
t['development_ratio_comedy_2'] = t['development_comedy_2'] / t['romantic_comedy_splited_length']

ttest_ind(t['development_ratio_comingofage_2'], t["development_ratio_comedy_2"])

Ttest_indResult(statistic=2.754326111904449, pvalue=0.006192127738204942)

In [50]:
Total_31 = t['development_comingofage_2'].mean()

Total_32 = t['development_comedy_2'].mean()
print(Total_31)
print(Total_32)

1.5632183908045978
0.6781609195402298
